src/routes/attestation.ts

```diff
@@ -1,21 +1,53 @@
-import { getBatch } from '../database.ts';
+import { FastifyInstance, FastifyRequest } from 'fastify';
+import constants from '../constants.js';
+import { deriveAddress } from '@blockfrost/blockfrost-js';
+import { blockfrostClient } from '../services/blockfrost.js';
+import { store } from '../services/database.js';
+import { AttestationQueryParameters } from '../types/routes.js';
+import { mnemonicToPrivateKey, getAccountKey, getXpub } from '../utils/keys.js';
+import { parseBatch } from '../utils/routes.js';
 
-export default (attestationId: string) => {
-  const attestation = getBatch(attestationId);
+async function attestation(fastify: FastifyInstance) {
+  fastify.route({
+    url: '/attestation/:order_id',
+    method: 'GET',
+    handler: async (request: FastifyRequest<AttestationQueryParameters>, reply) => {
+      const batch = await store.getBatch(request.params.order_id);
 
-  if (attestation) {
-    return Response.json({
-      order_id: attestation.uuid,
-      payment: {
-        address: attestation.address,
-        amount: attestation.amount,
-      },
-      status: attestation.status,
-      orderTimeLeftInSeconds: attestation.order_time_limit_in_seconds,
-    });
-  }
+      if (batch) {
+        const parsedBatch = parseBatch(batch);
+        return reply.send(parsedBatch);
+      }
 
-  return new Response(`Attestation with id: ${attestationId} was not found`, {
-    status: 404,
+      return reply.code(404).send('Not Found');
+    },
   });
-};
+
+  fastify.route({
+    url: '/attestation/submit',
+    method: 'POST',
+    handler: async (request: FastifyRequest<AttestationQueryParameters>, reply) => {
+      const addressIndex = await store.getBatchesCount();
+      const privateKey = mnemonicToPrivateKey(constants.mnemonic);
+      const accountKey = getAccountKey(privateKey);
+      const xpub = getXpub(accountKey);
+      const { address } = deriveAddress(
+        xpub,
+        1,
```

Comment:
most likely you want 0 for external addresses, 1 is reserved for external (change)